File: QueryOptimizer.Shared/Common/Models/ExecutionPlan/NormalizedExecutionPlan.cs
using QueryOptimizer.Shared.Common.Enums;
using System;
using System.Collections.Generic;
using System.Text;

namespace QueryOptimizer.Shared.Common.Models.ExecutionPlan
{
    public class NormalizedExecutionPlan
    {
        public DatabaseTypes Provider { get; set; }

        public string RawPlan { get; set; } = default!;

        public List<ExecutionPlanNode> Nodes { get; set; } = new();

        public double? TotalCost { get; set; }

        public double? TotalExecutionTimeMs { get; set; }

        public long? TotalLogicalReads { get; set; }

        public long? TotalPhysicalReads { get; set; }
    }
}

File: QueryOptimizer.API/app/query-optimizer-front/src/app/models/normalizedExecutionPlan.ts
import { ExecutionPlanNode } from './executionPlanNode';

export interface NormalizedExecutionPlan {
  provider: number;

  rawPlan?: string | null;

  nodes?: ExecutionPlanNode[];

  totalCost?: number | null;

  totalExecutionTimeMs?: number | null;

  totalLogicalReads?: number | null;

  totalPhysicalReads?: number | null;

  requestedMemoryKb?: number | null;

  grantedMemoryKb?: number | null;

  maxUsedMemoryKb?: number | null;

  additionalMetrics?: Record<string, string>;
}